src/excmds.ts

```diff
@@ -3335,6 +3340,72 @@ export async function ttscontrol(action: string) {
 
 //}}}
 
+// {{{ PERFORMANCE LOGGING
+
+/**
+ * Build a set of FilterConfigs from a list of human-input filter
+ * specs.
+ *
+ * @hidden
+ */
+//#background_helper
+export function buildFilterConfigs(filters: string[]): Perf.StatsFilterConfig[] {
+    return filters.map(
+        (filter: string): Perf.StatsFilterConfig => {
+            if (filter.endsWith("/")) {
+                return { kind: "ownerName", ownerName: filter.slice(0, -1) }
+            } else if (filter === ":start") {
+                return { kind: "eventType", eventType: "start" }
+            } else if (filter === ":end") {
+                return { kind: "eventType", eventType: "end" }
+            } else if (filter === ":measure") {
+                return { kind: "eventType", eventType: "measure" }
+            } else {
+                return { kind: "functionName", functionName: name }
+            }
+        },
+    )
+}
+
+/**
+ * Dump the raw json for our performance counters. Filters with
+ * trailing slashes are class names, :start | :end | :measure specify
+ * what type of sample to pass through, and all others are function
+ * names. All filters must match for a sample to be dumped.
+ *
+ * Tridactyl does not collect performance information by default. To
+ * get this data you'll have to set the configuration option
+ * `perfcounters` to `"true"`. You may also want to examine the value
+ * of `perfsamples`.
+ */
+//#background
+export async function dumpcounters(...filters: string[]) {
```

Comment:
Generally we try to pseudo-namespace ex commands with prefixes, e.g. tabclose / tabopen.

Could you rename this to perfdump / perfcounters?

Reply:
Heh, I _started_ with those but kept mistyping them because they were noun-verb instead of verb-noun. Give me a second to fix it.